unity-client/Assets/Scripts/MainScripts/DCL/Components/Avatar/AvatarShape.cs

```diff
@@ -189,5 +188,12 @@ public override void Cleanup()
             avatarUserInfo.userId = model.id;
             MinimapMetadataController.i?.UpdateMinimapUserInformation(avatarUserInfo, true);
         }
+
+        public override void SetModel(object model)
+        {
+            this.model = (AvatarModel)model;
+
+            //TODO (Adrian): This should handle the change of the model. This will required a major refactor so if you really need it, feel free to implement it!
+        }
```

Comment:
shouldn't we call this new `SetModel()` inside this class `ApplyChanges()` instead of the old `model = Utils.SafeFromJson<AvatarModel>(newJson);` ? it would be something like `SetModel(Utils.SafeFromJson<AvatarModel>(newJson));` I think.

Also, this new `SetModel` method across all the components is starting to look like something we would like to have in a parent class right? like inside `BaseComponent`

Reply:
Same as before, the `SetModel` call is designed to update the model component in the same frame so call to ApplyChanges won't work for that